package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+       *
+       * @private
+       */
+      _throwAbstractType: function() {
+        throw error.operInvalid(bundle.format(
+            bundle.structured.errors.instance.cannotCreateInstanceOfAbstractType, [this._getErrorLabel()]));
+      },
+
+      /**
+       * Gets a label suitable to identify this type in an error message.
+       *
+       * @return {string} An error label.
+       *
+       * @private
+       */
+      _getErrorLabel: function() {
+        return this.id || this.label; // Never empty;
+      },
+      //endregion
+
+      /**
+       * Determines if a value is an instance of this type.
+       *
+       * @param {?any} value The value to test.
+       * @return {boolean} `true` if the value is an instance of this type, `false` otherwise.
+       */
+      is: function(value) {
+        return O_isProtoOf.call(this.instance, value);
+      },
+
+      /**
+       * Determines if this is a subtype of another.
+       *
+       * A type is considered a subtype of itself.
+       *
+       * @param {?pentaho.type.Type} superType The candidate super-type.
+       * @return {boolean} `true` if this is a subtype of `superType` type, `false` otherwise.
+       */
+      isSubtypeOf: function(superType) {
+        return !!superType && (superType === this || O_isProtoOf.call(superType, this));
+      },
+
+      // TODO: is conversion always successful? If so, should it be?
+      /**
+       * Converts a value to an instance of this type,
+       * if it is not one already.
+       *
+       * If a {@link Nully} value is specified, `null` is returned.
+       *
+       * Otherwise, if a given value is not already an instance of this type
+       * (checked using [is]{@link pentaho.type.Type#is}),
+       * this method delegates the creation of an instance to
+       * [create]{@link pentaho.type.Type#create}.
+       *
+       * @param {?any} value The value to convert.
+       * @return {?pentaho.type.Instance} The converted value or `null`.
+       */
+      to: function(value) {
+        return value == null   ? null  :
+               this.is(value)  ? value :
+               this.create(value);
+      },
+
+      //region serialization
+      /**
+       * Creates a specification that describes this type.
+       *
+       * If an [ambient specification context]{@link pentaho.type.SpecificationContext.current},
+       * currently exists, it is used to manage the serialization process.
+       * Otherwise, one is created and set as current.
+       *
+       * This method creates a new {@link pentaho.type.SpecificationScope} for describing
+       * this type, and any other instances and types it references,
+       * and then delegates the actual work to {@link pentaho.type.Type#toSpecInContext}.
+       *
+       * @param {Object} [keyArgs] - The keyword arguments object.
+       * Passed to every type and instance serialized within this scope.
```

Comment:
Suggest rewording:

The keyword arguments object is passed to every type and instance serialized within this scope.